stores/moderationSubmissionsStore.ts

```diff
@@ -85,18 +87,25 @@ export const useModerationSubmissionsStore = defineStore(
             }
         }
 
-        async function updateSubmission(submission: MutationUpdateSubmissionArgs) {
-            try {
-                return await graphQLClientRequestWithRetry(
-                    gqlClient.request.bind(gqlClient),
-                    updateFacilitySubmissionGqlMutation,
-                    submission
-                )
-            } catch (error) {
-                console.error('Failed to update submission:', error)
+        async function updateSubmission(submission: MutationUpdateSubmissionArgs):
+        Promise<ServerResponse<Maybe<Submission>>> {
+            const serverResponse = { data: {} as Maybe<Submission>, errors: [] as ServerError[], hasErrors: false }
+            const response = await graphQLClientRequestWithRetry<Mutation>(
+                gqlClient.request.bind(gqlClient),
+                updateFacilitySubmissionGqlMutation,
+                submission
+            )
+
+            if (response.errors?.length) {
                 setDidMutationFail(true)
                 setUpdatingSubmissionFromTopBar(false)
             }
+
+            serverResponse.data = response.data?.updateSubmission
+            serverResponse.errors = response.errors ? response.errors : []
+            serverResponse.hasErrors = response.hasErrors
+
+            return serverResponse
```

Comment:
@ermish I had to change the way the function was written in order to show what type of mutation it was. And make a new object to send since we do a Mutation, and get back the type of updateSubmission. But the api call doesn't work this way. They can sese that it is updateSubmission on line 104. But their isn't really a way to access the key since the key as a type that I can see as that causes a mismatch of what we get

Reply:
If I try to specify the type I get this
![image](https://github.com/user-attachments/assets/d3e28051-4c5f-4913-ab8a-2b5cf5639579)
From trying the following code

```

        async function updateSubmission(submission: MutationUpdateSubmissionArgs):
        Promise<ServerResponse<Mutation['updateSubmission']>> {
            const response = await graphQLClientRequestWithRetry<Mutation['updateSubmission']>(
                gqlClient.request.bind(gqlClient),
                updateFacilitySubmissionGqlMutation,
                submission
            )

            if (response.errors?.length) {
                setDidMutationFail(true)
                setUpdatingSubmissionFromTopBar(false)
            }

            return response
        }
        ```

Reply:
I then cannot access this key due to typing

Reply:
you've done some good research on typing  😸  this is honestly an issue in pretty much every typed language. 

My advice at this point is it's more hassle to make it work for the nested Mutation type. (hence why it's not on the server side) 
I think it's fine to cast when accessing the inner object in the <mutation>, but the <serverResponse> and <Mutation> should at least be in the function's return type. So basically, `Promise<ServerResponse<Mutation>>` is good 👍 

Looking at what you got above, I think that is ideal ✨